app/controllers/course/stage/instructions.ts

```diff
@@ -47,16 +48,32 @@ export default class CourseStageInstructionsController extends Controller {
     return this.model.courseStage.prerequisiteInstructionsMarkdownFor(this.model.activeRepository);
   }
 
+  get shouldHideTestRunnerCardBeforeUserHasSubmitted() {
+    return this.featureFlags.hideTestRunnerCardBeforeUserHasSubmitted;
+  }
+
   get shouldShowFeedbackPrompt() {
     return !this.currentStep.courseStage.isFirst && this.currentStep.status === 'complete';
   }
 
+  get shouldShowLinkToForum() {
+    return (
+      this.isCurrentStage &&
+      this.currentStep.status !== 'complete' &&
+      this.currentStep.testsStatus !== 'passed' &&
+      this.shouldHideTestRunnerCardBeforeUserHasSubmitted
+    );
+  }
+
   get shouldShowPrerequisites() {
     return !!this.prerequisiteInstructionsMarkdown;
   }
 
   get shouldShowTestRunnerCard() {
-    return this.isCurrentStage && this.currentStep.status !== 'complete';
+    const shouldHideTestRunnerCard =
+      this.model.courseStage.isFirst && this.model.activeRepository.submissions.length <= 1 && this.shouldHideTestRunnerCardBeforeUserHasSubmitted;
```

Comment:
@andy1li we actually only load the "last" submission, so `submissions.length` isn't going to always be reliable. When the user is on the page, the older submissions will remain cached so it'll work but if the user refreshes, it'll be removed. Can we try using `repository.submissionsCount` instead?